osu.Game/Screens/Play/HUD/LegacyComboSplash.cs

```diff
@@ -0,0 +1,206 @@
+                    sprite.MoveToX(-sprite.Width * 0.625f).Then().MoveToX(0, 700, Easing.Out);
+                    sprite.FadeTo(1).Delay(200).FadeOut(1000, Easing.In);
+                }
+            }
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(ScoreProcessor scoreProcessor, ISkinSource skin)
+        {
+            Current.BindTo(scoreProcessor.Combo);
+
+            var c = new LegacyComboSplashComponent();
+
+            if (skin.GetDrawableComponent(c) is LegacyComboSplashSide side1)
+            {
+                side1.Origin = Anchor.CentreLeft;
+                side1.Anchor = Anchor.CentreLeft;
+                left = side1;
+                Add(left);
+            }
+            else
+                Add(left = new Container<Sprite>());
+
+            if (skin.GetDrawableComponent(c) is LegacyComboSplashSide side2)
+            {
+                side2.Origin = Anchor.CentreLeft;
+                side2.Anchor = Anchor.CentreRight;
+                side2.Scale = new Vector2(-1, 1);
+                right = side2;
+                Add(right);
+            }
+            else
+                Add(right = new Container<Sprite>());
+        }
+
+        private void OnSideChanged(Side side)
+        {
+            switch (side)
+            {
+                case Side.Both:
+                case Side.Random:
+                    // when both sides are used, the component should cover the entire screen.
+                    AutoSizeAxes = Axes.None;
+                    RelativeSizeAxes = Axes.Both;
+                    Position = new Vector2(0);
+                    Size = new Vector2(1, 1);
+                    break;
+
+                case Side.Left:
+                case Side.Right:
+                    RelativeSizeAxes = Axes.None;
+                    AutoSizeAxes = Axes.Both;
+                    break;
+            }
+
+            switch (side)
+            {
+                case Side.Both:
+                case Side.Random:
+                    left.FadeIn(500);
+                    right.FadeIn(500);
+                    break;
+
+                case Side.Left:
+                    left.FadeIn(500);
+                    right.FadeOut(500);
+                    break;
+
+                case Side.Right:
+                    left.FadeOut(500);
+                    right.FadeIn(500);
+                    break;
+            }
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Current.BindValueChanged(e => OnNewCombo(e.NewValue));
+            BurstsSide.BindValueChanged(e => OnSideChanged(e.NewValue), true);
+
+            if (Parent is SkinnableTargetComponentsContainer container)
+            {
+                container.ChangeChildDepth(this, 1000);
+            }
+        }
+
+        public enum Side
+        {
+            Left,
+            Right,
+            Both,
+            Random
```

Comment:
In stable both *is* random, for what it's worth. Not sure a new option needs to be invented here.